.travis.yml

```diff
@@ -1,17 +1,70 @@
+  access_key:
+    secure: OY3oWwiJghfty9wSPVvlhirvFGxPHDdIRuVkzAv6j7C/hj2BWYAP/UHrwdQ9XiYisHi/B5mGeyRVlrAf0MNGrG84rTDUbTWZbmktfuxl7A+Y6c0czk+s4SdhOiANG5b3tFl5wKq8h7uhrWH5/jWoKQ2Fz1VDCqxTvvZQbo41jSBhi7TBia626hxEePzdaiuw6HhGFZtfaoVs/FX30ylz8WDNrBjwCynjxsT52BaQrVvgEhuyzlOpI69YkZBPOq4fc3KiZ2YR43gLTx8K+sYCE9yJxdg1xT/UAawEhmedU83nyBZVo4rr7+03AixIxtI28MUCfBMlcsGwBxcKEKY/IWcp9UkPCq6+zALQoncV478tP21eYvlmxSFhYCrv+WEQlN+BcNjr4OJlmmFDbCVaF7r9qLeQPImU0+9iJU3OjrW7lpfLxORpGDEr2Nx6awKkIJCxNyK9weefeNo6Fz3V1kkyZ/7yWFeniJnRUCbahrB2XgzxIE+W307s1Qs4fm6JK7hVLTtG4fBzjChmAyGIzu744ws9WqmjvkC9D7OfnuXqanv/VcBFqPiudInerv7NL8FketUC+fxe/7XJfcxdaDGBjk8Kq7zXDohGRGymUXEoMDNJsKkMMlaKzdf7tgqdhsRJoH9NCVqrDXuG5al0UtrDP5RS7qfoxUunJmNFhlg=
+  no_ssl_bump_domains: companion.test
+
+matrix:
+  include:
+    - node_js: 'stable'
+      env: UNIT=true
+      addons:
+        apt: *APT
+    - node_js: 10
+      env: UNIT=true
+      addons:
+        apt: *APT
+    - node_js: 10
+      env: LINT=true
+      addons:
+        apt: *APT
+    - node_js: 10
+      env: BUILD=true
+      addons:
+        apt: *APT
+    - node_js: 10
+      env: COMPANION=true
+      addons:
+        apt: *APT
+    # Build the website when things are merged to master
+    - node_js: 10
+      env: WEBSITE=true
+      if: branch = master AND type = push
+      addons:
+        apt: *APT
+    # Run e2e tests on branch builds, not PRs
+    - node_js: 10
+      env: ENDTOEND=true
+      if: type = push
+      addons:
+        hosts: *HOSTS
+        apt: *APT
+        sauce_connect: *SAUCE
 before_install:
 - nvm install-latest-npm
 install:
 - npm install
 - npm run bootstrap -- --no-ci
 script:
-- npm run build
-- npm run test
-- if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then npm run test:endtoend; fi
+- 'if [ -n "${LINT-}" ]; then npm run lint; fi'
+- 'if [ -n "${LINT-}" ]; then npm run test:type; fi'
+- 'if [ -n "${WEBSITE-}" ]; then npm run web:install; fi'
+- 'if [ -n "${UNIT-}" ]; then npm run test:unit; fi'
+- 'if [ -n "${COMPANION-}" ]; then npm run test:companion; fi'
+- 'if [ -n "${ENDTOEND-}" ] && [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then npm run test:endtoend; fi'
+- 'if [ -n "${BUILD-}" ]; then npm run build; fi'
 - |
-  if [ "${TRAVIS_PULL_REQUEST}" = "false" ] &&\
+  if [ -n "${BUILD-}" ] && [ "${TRAVIS_PULL_REQUEST}" = "false" ] &&\
```

Comment:
Perhaps :- for travis pull request too?

Reply:
it's always available on travis, so i don't think that's necessary